From blp@nicira.com Thu Sep 10 23:25:10 2009
Received: from FTLPMAILMX02.citrite.net (10.9.154.224) by
	LONPMAILMX02.citrite.net (10.30.224.163) with Microsoft SMTP Server (TLS)
	id 8.1.393.1; Thu, 10 Sep 2009 23:25:10 +0100
Received: from SMTP02.CITRIX.COM (10.9.154.239) by FTLPMAILMX02.citrite.net
	(10.9.154.224) with Microsoft SMTP Server id 8.1.393.1; Thu, 10 Sep 2009
	18:25:08 -0400
Received: from mail.geoviz.com (HELO leadville.nicira.com)
	([209.172.104.10]) by SMTP02.CITRIX.COM with ESMTP; 10 Sep 2009 18:25:06
	-0400
Received: from hardrock.nicira.com ([172.16.0.20])	by leadville.nicira.com
	with esmtp (Exim 4.67)	(envelope-from <blp@nicira.com>)	id
	1Mls4a-0005v3-W2; Thu, 10 Sep 2009 15:25:00 -0700
Received: from blp by hardrock.nicira.com with local (Exim 4.67)
	(envelope-from <blp@hardrock.nicira.com>)	id 1Mls4e-0004UU-Se; Thu, 10 Sep
	2009 15:25:04 -0700
From: Ben Pfaff <blp@nicira.com>
To: "jpettit@nicira.com" <jpettit@nicira.com>, Ian Campbell <Ian.Campbell@eu.citrix.com>
CC: Ben Pfaff <blp@nicira.com>
Date: Thu, 10 Sep 2009 23:24:47 +0100
Subject: [PATCH 2/6] ovs-vsctl: Change "br-exists" exit code to 2 when the
	bridge doesn't exist.
Thread-Topic: [PATCH 2/6] ovs-vsctl: Change "br-exists" exit code to 2 when
	the bridge doesn't exist.
Thread-Index: AcoyZY8PfteG6wZcS5S69yKWBS5cwg==
Message-ID: <1252621491-17203-2-git-send-email-blp@nicira.com>
References: <1252621491-17203-1-git-send-email-blp@nicira.com>
In-Reply-To: <1252621491-17203-1-git-send-email-blp@nicira.com>
Accept-Language: en-GB, en-US
Content-Language: en-US
X-MS-Exchange-Organization-AuthAs: Anonymous
X-MS-Exchange-Organization-AuthSource: FTLPMAILMX02.citrite.net
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-sbrs: None
x-mesageid: 64739371
x-ironport-server: ftlpip02.citrite.net
x-remote-ip: 209.172.104.10
x-policy: $LOW_THROTTLE
x-ironport-anti-spam-filtered: true
x-ironport-anti-spam-result: AhkFAF8ZqUrRrGgK/2dsb2JhbACBU5UnyEiEGAU
x-ironport-av: E=Sophos;i="4.44,366,1249272000";   
	d="scan'208";a="64739371"
Content-Type: text/plain; charset="iso-8859-1"
MIME-Version: 1.0
X-Evolution-Source: imap://ianca@lonpmailmx01.citrite.net/
Content-Transfer-Encoding: 8bit

Exit code 1, previously used, is also used for other kinds of errors, so
it isn't a good choice.

Requested by Justin.
---
 tests/ovs-vsctl.at       |    2 +-
 utilities/ovs-vsctl.8.in |   10 +++++++++-
 utilities/ovs-vsctl.in   |    2 +-
 3 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/tests/ovs-vsctl.at b/tests/ovs-vsctl.at
index 28fc322..3cc1e92 100644
--- a/tests/ovs-vsctl.at
+++ b/tests/ovs-vsctl.at
@@ -12,7 +12,7 @@ m4_define([CHECK_BRIDGES],
      [m4_foreach([port], [$@], [port
 ])])
    m4_foreach([port], [$@], [AT_CHECK([RUN_OVS_VSCTL([br-exists port])])])
-   AT_CHECK([RUN_OVS_VSCTL([br-exists nonexistent])], [1])])
+   AT_CHECK([RUN_OVS_VSCTL([br-exists nonexistent])], [2])])
 
 m4_define([CHECK_PORTS],
   [AT_CHECK(
diff --git a/utilities/ovs-vsctl.8.in b/utilities/ovs-vsctl.8.in
index 7ed64da..587fa12 100644
--- a/utilities/ovs-vsctl.8.in
+++ b/utilities/ovs-vsctl.8.in
@@ -124,7 +124,7 @@ line.
 .IP "\fBbr-exists \fIbridge\fR"
 Tests whether \fIbridge\fR exists as a real or fake bridge.  If so,
 \fBovs\-vsctl\fR exits successfully with exit code 0.  If not,
-\fBovs-ctl\fR exits unsuccessfully with exit code 1.
+\fBovs-ctl\fR exits unsuccessfully with exit code 2.
 .
 .SS "Port Commands"
 .
@@ -157,6 +157,14 @@ more interfaces, rather than as a single port.
 Lists all of the interfaces within \fIbridge\fR on standard output,
 one per line.
 .
+.SH "EXIT STATUS"
+.IP "0"
+Successful program execution.
+.IP "1"
+Usage, syntax, or configuration file error.
+.IP "2"
+The \fIbridge\fR argument to \fBbr-exists\fR specified the name of a
+bridge that does not exist.
 .SH "SEE ALSO"
 .
 .BR ovs\-vswitchd.conf (5),
diff --git a/utilities/ovs-vsctl.in b/utilities/ovs-vsctl.in
index 3519b06..d51d922 100755
--- a/utilities/ovs-vsctl.in
+++ b/utilities/ovs-vsctl.in
@@ -361,7 +361,7 @@ def cmd_list_br():
 def cmd_br_exists(bridge):
     cfg = cfg_read(VSWITCHD_CONF)
     if bridge not in get_all_bridges(cfg):
-        sys.exit(1)
+        sys.exit(2)
 
 def cmd_list_ports(bridge):
     cfg = cfg_read(VSWITCHD_CONF)
-- 
1.6.3.3

